docs/resources/job.md

```diff
@@ -121,16 +121,18 @@ output "job_url" {
 The following arguments are required:
 
 * `name` - (Optional) An optional name for the job. The default value is Untitled.
-* `new_cluster` - (Optional) Same set of parameters as for [databricks_cluster](cluster.md) resource.
-* `existing_cluster_id` - (Optional) If existing_cluster_id, the ID of an existing [cluster](cluster.md) that will be used for all runs of this job. When running jobs on an existing cluster, you may need to manually restart the cluster if it stops responding. We strongly suggest to use `new_cluster` for greater reliability.
+* `job_clusters` - (Optional) A list of job [databricks_cluster](cluster.md) specifications that can be shared and reused by tasks of this job. Libraries cannot be declared in a shared job cluster. You must declare dependent libraries in task settings.
+* `new_cluster` - (Optional) Same set of parameters as for [databricks_cluster](cluster.md) resource. This is single-task (legacy) syntax
+* `existing_cluster_id` - (Optional) If existing_cluster_id, the ID of an existing [cluster](cluster.md) that will be used for all runs of this job. When running jobs on an existing cluster, you may need to manually restart the cluster if it stops responding. We strongly suggest to use `new_cluster` for greater reliability. This is single-task (legacy) syntax
```

Comment:
maybe we need to specify which options are relevant for "legacy" and what could be used only with multiple tasks